### PR TITLE
new package: redis

### PR DESCRIPTION
**Pull request by bhatiaharsh**
Spack package for Redis (https://redis.io)

### CODE REVIEW

File: var/spack/repos/builtin/packages/redis/package.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    """Redis is an open source (BSD licensed), in-memory data structure store,
+       used as a database, cache and message broker."""
+
+    homepage = "https://redis.io/"
+    url      = "http://download.redis.io/releases/redis-4.0.11.tar.gz"
+
+    version('4.0.11', 'a13ccf0f7051f82dc1c979bd94f0b9a9ba039122')
+    version('4.0.8',  'f723b327022cef981b4e1d69c37a8db2faeb0622')
+
+    def install(self, spec, prefix):
```
Comment (healther) [inline var/spack/repos/builtin/packages/redis/package.py]: this shouldn't be needed, MakefilePackage should take care of all these steps

File: var/spack/repos/builtin/packages/redis/package.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
```
Comment (healther) [inline var/spack/repos/builtin/packages/redis/package.py]: Please remove the boilerplate code

File: var/spack/repos/builtin/packages/redis/package.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+class Redis(MakefilePackage):
+    """Redis is an open source (BSD licensed), in-memory data structure store,
+       used as a database, cache and message broker."""
+
+    homepage = "https://redis.io/"
+    url      = "http://download.redis.io/releases/redis-4.0.11.tar.gz"
+
+    version('4.0.11', 'a13ccf0f7051f82dc1c979bd94f0b9a9ba039122')
+    version('4.0.8',  'f723b327022cef981b4e1d69c37a8db2faeb0622')
+
```
Comment (healther) [inline var/spack/repos/builtin/packages/redis/package.py]: are you sure that there are no dependencies needed?

File: var/spack/repos/builtin/packages/redis/package.py (hunk 1 above)
Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/redis/package.py]: # Redis make file uses a not standard flag to pass the install path. Hence I am overriding the install function.

File: var/spack/repos/builtin/packages/redis/package.py (hunk 2 above)
Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/redis/package.py]: Done

File: var/spack/repos/builtin/packages/redis/package.py (hunk 3 above)
Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/redis/package.py]: I am able to build Redis and use it as a dependency for my other tools, and it works fine.

File: var/spack/repos/builtin/packages/redis/package.py (hunk 1 above)
Comment (healther) [reply var/spack/repos/builtin/packages/redis/package.py]: Ah I see. I think then you only need to do 
```
def install(self, spec, prefix):
      make('PREFIX={0}'.format(prefix), 'install')
```
the `make()` step should be done by the `build` method of the `MakefilePackage` class.

Comment (bhatiaharsh) [reply var/spack/repos/builtin/packages/redis/package.py]: Okay. done. Thanks for the suggestion.

**Comment by michaelkuhn** [comment]
Seems to work for me. It is worth noting that redis seems to depend on (and bundle) jemalloc, lua, hiredis and linenoise. I am not sure whether it makes sense to try to unbundle some of those. Even Fedora only seems to unbundle jemalloc: https://src.fedoraproject.org/rpms/redis/blob/master/f/redis.spec

**Comment by scheibelp** [comment]
This looks good but can you update the license according to https://github.com/spack/spack/pull/9450#issue-221027331? The new license header comment looks like:

```
# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
# Spack Project Developers. See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: (Apache-2.0 OR MIT)
```

Spack changed its licensing recently.

**Comment by scheibelp** [comment]
This looks good but can you update the license according to https://github.com/spack/spack/pull/9450#issue-221027331? The new license header comment looks like:

```
# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
# Spack Project Developers. See the top-level COPYRIGHT file for details.
# 
# SPDX-License-Identifier: (Apache-2.0 OR MIT)
```

Spack changed its licensing recently.

**Comment by alalazo** [comment]
@scheibelp @tgamblin Just asking out of sheer legal ignorance: can we update the license header in cases like this or is it on the poster to actively change the license of the PR?

**Comment by koning** [comment]
@bhatiaharsh Can this be updated with the newest stable version 5 release?
version('5.0.3', 'aee2e1d0e50f87d6c21abef1c45f9767')

**Pull request closed without merging**